docs/api/connection.rst

```diff
@@ -339,10 +339,339 @@ Connection
-
 .. _BigInt:
     https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt
+
+
+.. _edgedb-js-api-pool:
+
+Pool
+==========
+
+.. js:function:: createPool(options)
+                 createPool(options, callback)
+
+    Create a connection pool to an EdgeDB server.
+
+    :param options: Connection pool parameters object.
+
+    :param ConnectConfig options.connectOptions:
+        Connection parameters object, used when establishing new connections.
+        Refer to the documentation at :ref:`edgedb-js-api-connection`.
+
+    :param number options.minSize:
+        The minimum number of connections initialized by the connection pool.
+        If not specified, this value is by default 0: the first connection is
+        created when required.
+
+    :param number options.maxSize:
+        The maximum number of connections created by the connection pool.
+        If not specified, this value is by default 100.
+
+    :param func options.onAcquire:
+        Optional callback, called when a connection is acquired.
+        *(proxy: PoolConnectionProxy) => Promise<void>*
+
+    :param func options.onRelease:
+        Optional callback, called when a connection is released.
+        *(proxy: PoolConnectionProxy) => Promise<void>*
+
+    :param func options.onConnect:
+        Optional callback, called when a new connection is created.
+        *(connection: AwaitConnection) => Promise<void>*
+
+    :param func options.connectionFactory:
+        Optional function, used to obtain a new connection. By default, the
+        function is :js:func:`connect` *(options?: ConnectConfig) =>
+        Promise<AwaitConnection>*
+
+    :param callback:
+        A callback function that will be invoked when the connection pool is
+        ready. The callback function should be of the form ``function(error,
+        pool)``. The *pool* is an instance of :js:class:`CallbackPool`.
+
+    :returns:
+        There are two ways of creating an EdgeDB connection pool: a
+        Promise-based approach and a callback-based. When a *callback*
+        argument is provided, the function does not return anything and instead
+        uses the *callback* with an instance of :js:class:`CallbackPool`.
+        Otherwise, a ``Promise`` of an :js:class:`Pool` is returned.
+
+.. js:class:: Pool
+
+    A connection pool is used to manage a set of connections to a database.
+    Since opening connections is an expensive operation, connection pools are
+    used to maintain and reuse connections, enhancing the performance of
+    database interactions.
+
+    Pools can be created using the method ``createPool``:
+
+    .. code-block:: js
+
+        const edgedb = require("edgedb");
+
+        async function main() {
+            const pool = await edgedb.createPool({
+                connectOptions: {
+                    user: "edgedb",
+                    host: "127.0.0.1",
+                },
+            });
+
+            await pool.initialize();
```

Comment:
Why is this necessary? Why can't `createPool` take care of this?

Reply:
Although it seems that `initialize()` is now protected and this looks like a line we just need to drop, right?

Reply:
Yes, it's a mistake.